File: competition/01.周赛/336.场周赛/02.. 重排数组以得到最大前缀分数.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var maxScore = function (nums) {
  nums.sort((a, b) => b - a)
  let prev = 0
  let count = 0
  for (let i = 0; i < nums.length; i++) {
    prev += nums[i]
    if (prev > 0) count++
    else break
  }
  return count
}
console.log(maxScore([2, -1, 0, 1, -3, 3, -3]), maxScore([-2, -3, 0]))

File: competition/01.周赛/336.场周赛/02.重排数组以得到最大前缀分数.java
class Solution {
    public int maxScore(int[] nums) {
 Arrays.sort(nums);
        long prev =0;
        int sum = 0;
        for (int i = nums.length - 1; i >= 0; i--) {
            prev += nums[i];
            if (prev > 0) sum++;
            else break;
        }
        return sum;
    }
}